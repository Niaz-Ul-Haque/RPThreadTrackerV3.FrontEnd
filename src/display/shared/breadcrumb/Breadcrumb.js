// #region imports
import React from 'react';
import { Route } from 'react-router-dom';
import Breadcrumbs from './Breadcrumbs';
// #endregion imports

export default props => (
	<div>
		<Route path="/:path" component={Breadcrumbs} {...props} />
	</div>
);
