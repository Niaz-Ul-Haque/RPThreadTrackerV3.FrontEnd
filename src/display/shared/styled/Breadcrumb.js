import styled from 'styled-components';
import { Breadcrumb } from 'reactstrap';
import colors from '../../../infrastructure/constants/colors';

export default styled(Breadcrumb)`
	border-bottom: none;
	margin-bottom: 0;
	background-color: ${colors.GRAY_900}
	a {
		color: ${colors.LIGHT_BLUE}
	}
	.light-theme & {
		background-color: ${colors.GRAY_100}
		a {
			color: ${colors.DARK_BLUE}
		}
	}
`;
