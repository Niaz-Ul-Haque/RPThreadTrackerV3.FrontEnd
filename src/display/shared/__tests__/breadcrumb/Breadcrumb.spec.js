// #region imports
import React from 'react';
import { shallow } from 'enzyme';
import Breadcrumb from '../../breadcrumb/Breadcrumb';
// #endregion imports

jest.mock('../../breadcrumb/Breadcrumbs', () => () => 'Breadcrumbs');

const createTestProps = propOverrides => ({
	...propOverrides
});
describe('rendering', () => {
	it('should render valid snapshot', () => {
		const props = createTestProps();
		const jsx = (<Breadcrumb {...props} />);
		const element = shallow(jsx);
		expect(element).toMatchSnapshot();
	});
	it('should pass props to child', () => {
		const props = createTestProps({ 'data-random': 'random' });
		const jsx = (<Breadcrumb {...props} />);
		const element = shallow(jsx);
		const route = element.find('Route');
		expect(route).toHaveProp('data-random', 'random');
	});
});
