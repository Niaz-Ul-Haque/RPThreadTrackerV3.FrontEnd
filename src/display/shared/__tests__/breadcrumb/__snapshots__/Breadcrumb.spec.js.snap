// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering snapshots should render valid snapshot 1`] = `
<div>
  <Route
    component={[Function]}
    path="/:path"
  />
</div>
`;
